test/CMakeLists.txt

```diff
@@ -88,19 +88,21 @@ function(add_test)
   # Call the original add_test
   _add_test(NAME ${test_name} COMMAND ${target})
 
-  set_property(TEST ${test_name} PROPERTY RUN_SERIAL YES)
+  set(pre_parent_path "$<1:$><PATH:GET_PARENT_PATH$<COMMA>")
+  set(post_parent_path "$<ANGLE-R>")
+  set(parent_path_glue "${post_parent_path}$<SEMICOLON>${pre_parent_path}")
 
-  if(TEST_PROPERTIES)
-    set_tests_properties(${test_name} PROPERTIES ${TEST_PROPERTIES})
-  endif()
+  set(dlls "$<TARGET_RUNTIME_DLLS:${target}>")
 
-  set(dll_command ${CMAKE_COMMAND} -E copy_if_different
-      $<TARGET_RUNTIME_DLLS:${target}> $<TARGET_FILE_DIR:${target}>)
-  add_custom_command(
-    TARGET ${target} POST_BUILD
-    COMMAND "$<$<BOOL:$<TARGET_RUNTIME_DLLS:${target}>>:${dll_command}>"
-    COMMAND_EXPAND_LISTS
+  set(directories
+      "$<SHELL_PATH:$<REMOVE_DUPLICATES:$<GENEX_EVAL:${pre_parent_path}$<JOIN:${dlls},${parent_path_glue}>${post_parent_path}>>>"
   )
+  set(prepend_path "PATH=path_list_prepend:")
+  set(path_modification
+      $<$<BOOL:${dlls}>:${prepend_path}$<JOIN:${directories},;${prepend_path}>>)
```

Comment:
Hmm, okay, maybe this genex soup isn't the best idea. I think maybe this sould be left for another PR for now and instead just the CI could be adjusted by editing the PATH in the yaml [like this](https://github.com/friendlyanon/cmake-init/blob/master/.github/workflows/ci.yml#L110-L112).

Reply:
Yes, it is a bit complicated. Basically what it is doing is using the `$<JOIN:...>` genex to apply the `$<PATH:GET_PARENT_PATH,...>` genex to every DLL (which is a little complicated, because you have to delay the later to run after the former), and then using removing duplicate directories in that resulting list, and then rejoining with another "$<JOIN:...>` genex to add the `PATH=path_list_prepend:` for the environment append.

Reply:
I'm really not a cmake expert, if those lines can be written with simpler if/else structure, please consider using that instead.

Reply:
The issue is when building as a shared library on Windows, the test executable won't see the DLL, because it will be in a different directory. My opinion is that for CI `PATH` can be appended and a developer locally could just pass `CMAKE_RUNTIME_OUTPUT_DIRECTORY` from the command line to put them all in one place or anything they wish to really.